external-crates/move/move-compiler/src/expansion/translate.rs

```diff
@@ -122,6 +126,40 @@ impl<'env, 'map> Context<'env, 'map> {
     }
 }
 
+/// We mark named addresses as having a conflict if there is not a bidirectional mapping between
+/// the name and its value
+fn compute_address_conflicts(
+    pre_compiled_lib: Option<&FullyCompiledProgram>,
+    prog: &P::Program,
+) -> BTreeSet<Symbol> {
+    let mut name_to_addr: BTreeMap<Symbol, BTreeSet<AccountAddress>> = BTreeMap::new();
+    let mut addr_to_name: BTreeMap<AccountAddress, BTreeSet<Symbol>> = BTreeMap::new();
+    let all_addrs = prog.named_address_maps.all().iter().chain(
+        pre_compiled_lib
+            .iter()
+            .flat_map(|pre| pre.parser.named_address_maps.all()),
+    );
+    for map in all_addrs {
+        for (n, addr) in map {
+            let n = *n;
+            let addr = addr.into_inner();
+            name_to_addr.entry(n).or_default().insert(addr);
+            addr_to_name.entry(addr).or_default().insert(n);
+        }
+    }
+    let name_to_addr_conflicts = name_to_addr
+        .into_iter()
+        .filter(|(_, addrs)| addrs.len() > 1)
+        .map(|(n, _)| n);
```

Comment:
Nit: consider a filter_map here and below instead
```suggestion
        .filter_map(|(n, addrs)| if addrs.len() > 1 { Some(n) } else { None });
```

Reply:
I feel that is less readable than a single filter+map. I tend to only reach for filter_map if I'm already dealing with options, or have many intermediate steps.